### DISCUSSION TITLE
MDX documentation ArgsTable: "Args table with interactive controls couldn't be auto-generated"

### DISCUSSION THREAD

**Discussion by bobrovnikovaSelfridges** [storybookjs/storybook#23959] [Help]
Facing the same issue in almost every somehow unique component. 
<img width="834" alt="image" src="https://github.com/storybookjs/storybook/assets/103778870/2d12087c-0d80-48a6-a182-56fc99190de2">

All possible args are enabled. Tried: with arg, with argTypes, with both of them, without any at all - still doesn't work. Tried to remove and add each after another argument - no results. Tried to define types both in mdx and in story.
Code:
```
<ArgsTable of={RadioButtonStory} />

<Meta
  title='RadioButton doc'
  of={RadioButtonStory}
  argTypes={ 
    {name: {
      control: 'text',
     },
    options: {
      control: 'object',
      description: 'An array of options here.',
    },
    onChange: {
       action: 'changed',
      description: 'Callback here',
    },
    orientation: {
      control: 'radio',
      options: ['A', 'B'],
      description: 'ABC...',
    }}}
/>
```

Even the message confuses me -- Args table with interactive controls couldn't be auto-generated? But why for some components it works? What does it mean **interactive controls**? I thought all controls are interactive. 
Would be glad to get an advice <3

**Comment by lqzhgood** [comment]
maybe about to https://github.com/storybookjs/storybook/discussions/24618 ?

**Comment by yuchiu** [comment]
Try installing this package `@storybook/components`, or update it to `@storybook/components@7.6.17` if it's an older version. This resolves my issue.

https://github.com/storybookjs/storybook/discussions/24738#discussioncomment-8954305

**Comment by centigrade-julian-lang** [comment]
For me it was stupidly simple - the table wasn’t generated because there was nothing to document.

The component behind the story exposed no public inputs and outputs and the referenced story’s `meta.args` was empty. As a result, Storybook had no data to render for `<ArgTypes />`. Although there were public methods (some Angular life-cycle hooks), it _seems_ like compodoc needs at least an input or output in order to list the public life-cycle hooks of angular as well.

I use Compodoc and confirmed it by looking at the component’s entry in `documentation.json` - it contained no inputs (`inputsClass` property) or outputs (`outputsClass` property), so the table could not be generated.

Example:

```html
<!-- my-story.mdx -->
import { Meta, ArgTypes } from '@storybook/blocks';
import * as MyStories from '../stories/my-story';

<Meta of={MyStories} />
<ArgTypes />
```

```ts
@Component({ ... })
export class MyStoryComponent implements OnInit, OnDestroy {
   // no inputs, outputs and public class members besides the angular hooks
   ngOnInit() { ... }

   ngOnDestroy() { ... }
}
```

```ts
// my.stories.ts
const meta: Meta<MyStory> = {
  component: MyStoryComponent,
  args: {}, // nothing to document
};

export default meta;
```

**Reply by centigrade-julian-lang** [reply]
Also make sure that your components are actually included in your storybook's `tsconfig.json`, just happened again - the `documentation.json` was completely empty, because the `.storybook/tsconfig.json#include` path was broken ;)
